### PR TITLE
Add EsslShaderGenerator and Three.js based sample app (JsMaterialXView)

### PR DESCRIPTION
**Pull request by muenstc**
Does not include the javascript bindings for the EsslShaderGenerator (this will be a follow up PR).
WebGL2 compatible shaders can be generated from the MaterialXView app by pressing `E`.
JsMaterialXView renders the default standard surface material using generated shaders.

### CODE REVIEW

**Review by sdunkel — APPROVED** [review]
Just some nits, otherwise LGTM

File: libraries/pbrlib/genglsl/lib/mx_microfacet_specular.glsl [hunk 1]
```diff
@@ -233,12 +233,12 @@ vec3 mx_depolarize(vec3 s, vec3 p)
 vec3 mx_eval_sensitivity(float opd, float shift)
 {
     // Use Gaussian fits, given by 3 parameters: val, pos and var
-    float phase = 2*M_PI * opd;
+    float phase = 2.0*M_PI * opd;
```
[the remark this answers is not in the document]
Comment (muenstc) [reply libraries/pbrlib/genglsl/lib/mx_microfacet_specular.glsl:236]: If we do this we should have a pass over all the shader snippets. I assume we will find some more occurrences of such issues. Not just in the snippets we adapted for this PR.

File: resources/Materials/Examples/StandardSurface/user_bug.mtlx [hunk 2]
```diff
@@ -0,0 +1,24 @@
+<?xml version="1.0"?>
```
[the remark this answers is not in the document]
Comment (muenstc) [reply resources/Materials/Examples/StandardSurface/user_bug.mtlx:1]: Nope. Seems like a leftover. Good catch!

File: source/MaterialXGenEssl/EsslShaderGenerator.h [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+{
+  public:
+    EsslShaderGenerator();
+
+    static ShaderGeneratorPtr create() { return std::make_shared<EsslShaderGenerator>(); }
+
+    /// Return a unique identifier for the target this generator is for
+    const string& getTarget() const override { return TARGET; }
+
+    const string& getVersion() const override { return VERSION; }
```
[the remark this answers is not in the document]
Comment (muenstc) [reply source/MaterialXGenEssl/EsslShaderGenerator.h:31]: Those are just overrides.

File: libraries/pbrlib/genglsl/lib/mx_microfacet_specular.glsl (hunk 1 above)
[the remark this answers is not in the document]
Comment (sdunkel) [reply libraries/pbrlib/genglsl/lib/mx_microfacet_specular.glsl:236]: ok, then this should be done in a separate PR

**Pull request merged**